onnxruntime/core/platform/windows/env.cc

```diff
@@ -459,8 +459,8 @@ Status WindowsEnv::MapFileIntoMemory(_In_z_ const ORTCHAR_T* file_path,
 
   void* const mapped_base = MapViewOfFile(file_mapping_handle.get(),
                                           FILE_MAP_READ,
-                                          0,
-                                          static_cast<DWORD>(mapped_offset),
+                                          static_cast<DWORD>((mapped_offset >> 32) & 0xFFFFFFFF),
+                                          static_cast<DWORD>(mapped_offset & 0xFFFFFFFF),
```

Comment:
Should we use 0xFFFFFFFFul like this answer suggests? https://stackoverflow.com/a/9894259

Reply:
I think it is not needed. Let me lookup C/C++ manual now.

Reply:
C++ standard says this is a hexadecimal literal. A compiler will try to find the first of the following types that its value can be represented.

1. int
2. unsigned int
3. long int
4. unsigned long int
5. long long int
6. unsigned long long int

Apparently int is not ok,  so it will use "unsigned int".  If without the "0x" prefix, the type would be a signed type.

Reply:
Thanks for looking that up. Would it be clearer to use the ul suffix? Then we don't need to remember the order defined by the standard and will be helpful for future maintainers by just looking at this line.

Reply:
I think the problem is elsewhere. mapped_offset  is a signed 64-bit value. 0xFFFFFFFF is unsigned 32-bit.  The result type of "mapped_offset & 0xFFFFFFFF" should be int64_t, then we cast it back to DWORD is a 32-bit unsigned integer.